### ISSUE TITLE
changetype should not be a required field in the openapi specification

### ISSUE THREAD

**Issue by marcel-hamer** [PowerDNS/pdns#14052]
- [x] This is not a support question, I have read [about opensource](https://www.powerdns.com/opensource.html) and will send support questions to the IRC channel, [Github Discussions](https://github.com/PowerDNS/pdns/discussions/) or the mailing list.
- [x] I have read and understood the ['out in the open' support policy](https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open/)

 - Program: Authoritative
 - Issue type: Bug report

### Short description
In the current [authoritative-api-swagger.yaml](https://github.com/PowerDNS/pdns/blob/617d2f04de5b3fafb173541ee1933c2a8f4a3347/docs/http-api/swagger/authoritative-api-swagger.yaml#L1075) specification the changetype is listed as required for rrsets. But when calling the endpoint for [requesting a zone managed by a server](https://github.com/PowerDNS/pdns/blob/617d2f04de5b3fafb173541ee1933c2a8f4a3347/docs/http-api/swagger/authoritative-api-swagger.yaml#L169) and including rrsets, the returned rrsets do not include the entry changetype.

As such I think the changetype field cannot be declared as _required_.

### Environment

 - Operating system: Linux
 - Software version: auth-4.9.0
 - Software source: PowerDNS repository

### Steps to reproduce
1. 
```
curl -X 'GET' \
  'https://<powerdns-api-server>/api/v1/servers/<host>/zones/<zonename>?rrsets=true' \
  -H 'accept: application/json' \
  -H 'X-API-Key: <api-key>'
```
2. Check the json output and then it can be seen 'changetype' is missing for rrsets.

### Expected behaviour

I think the output is fine, but it doesn't match the specification because of 'changetype' being marked as required.

### Actual behaviour

-

### Other information

-

**Comment by kpfleming** [comment]
Good catch! The proper solution to this problem is to make two types for RRSets; one for change requests and one for responses, with the `changetype` field only in the request version.

**Comment by miodvallat** [comment]
This has been fixed as part of #12983.
